### PR TITLE
feat(menu-item): add max width prop for truncation

### PR DESCRIPTION
**Pull request by sjohnson-sage**
### Proposed behaviour
Add a new `maxWidth` prop to `MenuItem`. If the title of the item exceeds the maximum width, it will automatically truncate.
This will also add a title attribute for hovering over the `MenuItem`.

image

### Current behaviour
It is not currently possible to add truncation to `MenuItem` titles in this way.

### Checklist

- [x] Commits follow our style guide
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [ ] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Typescript `d.ts` file added or updated if required
- [x] Carbon implementation and Design System documentation are congruent

### Testing instructions
https://codesandbox.io/s/fervent-noyce-y162u

### CODE REVIEW

File: src/components/menu/menu-item/menu-item.component.js [hunk 1]
```diff
@@ -129,6 +132,7 @@ const MenuItem = ({
       menuType={menuContext.menuType}
       inSubmenu={submenuContext.handleKeyDown !== undefined}
       display="inline-block"
+      title={maxWidth && typeof children === "string" ? children : ""}
```
Comment (mmaciorowski-sage) [inline src/components/menu/menu-item/menu-item.component.js:135]: maybe you should create a function?
some like:
`const getTitle = title => maxWidth && typeof title === "string" ? title : ""`

**Review by edleeks87 — DISMISSED** [review]

File: src/components/menu/__internal__/submenu/submenu.component.js [hunk 2]
```diff
@@ -322,6 +324,8 @@ Submenu.propTypes = {
   clickToOpen: PropTypes.bool,
   /** The href to use for the menu item. */
   href: PropTypes.string,
+  /** Maximum width in px */
```
Comment (jamime) [inline src/components/menu/__internal__/submenu/submenu.component.js:327]: I think we could relax this to allow any CSS string.

**Review by edleeks87 — APPROVED** [review]

**Review by mmaciorowski-sage — APPROVED** [review]

File: src/components/menu/__internal__/submenu/submenu.component.js [hunk 3]
```diff
@@ -322,6 +324,8 @@ Submenu.propTypes = {
...
+  maxWidth: PropTypes.number,
```
Comment (jamime) [inline src/components/menu/__internal__/submenu/submenu.component.js:328]:
```suggestion
  maxWidth: PropTypes.string,
```

File: src/components/menu/__internal__/submenu/submenu.component.js (hunk 2 above)
Comment (jamime) [inline src/components/menu/__internal__/submenu/submenu.component.js:327]:
```suggestion
```

**Review by jamime — CHANGES_REQUESTED** [review]

File: src/components/menu/__internal__/submenu/submenu.component.js (hunk 2 above)
Comment (sjohnson-sage) [reply src/components/menu/__internal__/submenu/submenu.component.js:327]: 🤦🏻

**Review by jamime — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by mmaciorowski-sage — APPROVED** [review]

**Review by KatarzynaQA — APPROVED** [review]

**Pull request merged**